app/forms/hub/client_form.rb

```diff
@@ -76,16 +77,26 @@ def dependent_validation_context
     end
 
     def opted_in_sms?
-      sms_notification_opt_in == "yes"
+      sms_notification_opt_in == "yes" && sms_phone_number.present?
     end
 
     def opted_in_email?
-      email_notification_opt_in == "yes"
+      email_notification_opt_in == "yes" && email_address.present?
     end
 
     def at_least_one_contact_method
       unless opted_in_email? || opted_in_sms?
-        errors.add(:communication_preference, I18n.t("forms.errors.need_one_communication_method"))
+        errors.add(:email_address, I18n.t("forms.errors.hub.communication_opt_in"))
+        errors.add(:sms_phone_number, I18n.t("forms.errors.hub.communication_opt_in"))
+      end
+    end
+
+    def contact_method_if_opted_in
```

Comment:
rename to `validate_notification_opt_ins` maybe

Reply:
`validate :opt_in_checkboxes`
`validate :opt_in_checkboxes_have_corresponding_information`